src/python/pants/engine/internals/parametrize_test.py

```diff
@@ -126,6 +126,25 @@ def test_expand(
         key=lambda value: value[0],
     )
 
+@pytest.mark.parametrize(
+    "expected,parameters,fields",
+    [
+        ([("a@g=c", {"f": "b"})], {"g": "c"}, {"f": "b"}),
+        ([("a@f=b,g=c", {"f": "b"})], {"g": "c"}, {"f": Parametrize("b")}),
+        ([("a@f=b", {"f": "b"})], {"f": "a"}, {"f": "b"}),
```

Comment:
I think this wrong. I think this overrides the parametrization and goes back to not being parametrized, so the resulting address should just be `"a"`.

Reply:
I'm not sure at what abstraction level it belongs. I could remove the parameter field from the address instead of adding a new single parameter, but should this not be handled the same way than above?

In other word, the test `([("a@f=b,g=c", {"f": "b"})], {"g": "c"}, {"f": Parametrize("b")}),` should also be replaced to `([("a@g=c", {"f": "b"})], {"g": "c"}, {"f": Parametrize("b")}),`?

Same for `([("a", {"f": "b"})], {"f": "a"}, {"f": Parametrize("b")}),`?

In short, using `Parametrize(x)` anywhere should be equivalent to using `x` alone?

Reply:
> using `Parametrize(x)` anywhere should be equivalent to using `x` alone?

I see your point, and it's a good one and other may agree. For me, I'm inclined to argue for it still being a parametrized target albeit with only a single value. But for whatever reason I feel like we want to distinguish the two cases (of a field value with/without the `parametrize()` wrapped around it.)

cc @pantsbuild/maintainers for more 👀 on direction here..

Context: When we use `parametrize()` on a target, those fields are added to the target's address on the form `tgt-name@field1=alias1,field2=alias2` for each field that is parametrized.

@isra17 discovered that we fail to update these address parameters (the part after `@`) for fields set using the `overrides` field, and in fixing this (in this PR) I feel that if the new value is _no longer_ a parametrized value, we should remove it from the address parameters.

The question then becomes to fix this generally and not only for this case, and whether to treat a parametrized field with only a single value as parametrized at all (as there will still only be a single version of the target, based on that parametrization).